samples/src/JustSaying.Sample.Restaurant.KitchenConsole/Program.cs

```diff
@@ -98,6 +101,44 @@ private static async Task Run()
                             x.ForTopic<OrderOnItsWayEvent>(cfg =>
                                 cfg.WithReadConfiguration(rc =>
                                     rc.WithSubscriptionGroup("GroupB")));
+
+
+                            // TODO - REVERT BEFORE MERGING
+                            // New proposal
+                            // Option 1
+                            x.ForQueue<OrderPlacedEvent>(QueueAddress.FromArn("arn:aws:sqs:eu-west-1:111122223333:queue1"));
+                            // Option 2
+                            x.ForQueueArn<OrderPlacedEvent>("arn:aws:sqs:eu-west-1:111122223333:queue1");
+
+                            // To revert to the classic "auto-infrastructure" mode
+                            // Only possible with Option 1
+                            x.ForQueue<OrderPlacedEvent>(QueueAddress.None);
+
+                            // try the simplified builder
+                            // Option 1
+                            x.ForQueue<OrderPlacedEvent>(QueueAddress.FromArn("arn:aws:sqs:eu-west-1:111122223333:queue1"), cfg => { });
+                            // Option 2
+                            x.ForQueueArn<OrderPlacedEvent>("arn:aws:sqs:eu-west-1:111122223333:queue1", cfg => { });
```

Comment:
@martincostello I've added top-level APIs (Option 2 here).
We could go either way with this, as it's mostly an aesthetic change.

Option 1 Pros:
Having a `QueueAddress.None` is handy, as we can use it as a signal to use the classic auto-generated infra for convenience, so going between the modes feels a little simpler rather than calling 2 different overloads (minor nit IMO)

Option 2 Pros:
Less method overload headache for consumers.
Shorter registration lines.

Reply:
Out of the two on pure aesthetics, I'd go with 2.